### PR TITLE
Fixed impossibility to search for vouchers in the voucher list table

### LINKED ISSUE #2005: Voucher cannot be searched for on the vouchers page
### Reproduction Steps

Go to your dashboard 
Go to the vouchers page
Try searching for a voucher

### Expected Behavior

It should search and return the voucher that contains the search field

### Actual Behavior

It gives this error:
image

### Error Messages / Logs

+ Include any relevant error messages or log files
```
N/A
```

### System and Environment Information

<details>
<summary>System Report</summary>

```
Wordpress
-------------------------------------------

Home Url: [removed]
Site Url: [removed]
Login Url: [removed]/wp-login.php
Version: 5.9
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Home (#889) [[removed]/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 5.9.0
Db Version: 5.9.0
Course Catalog: Courses (#245) [[removed]/courses-2/]
Membership Catalog: Membership Catalog (#6) [[removed]/memberships/]
Student Dashboard: Dashboard (#8) [[removed]/dashboard/]
Checkout Page: Purchase (#7) [[removed]/purchase/]
Course Catalog Per Page: -1
Course Catalog Sorting: menu_order,ASC
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order,ASC
Site Membership: Not Set
Courses Endpoint: %e3%82%b3%e3%83%bc%e3%82%b9
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: no
Password Strength Meter: yes
Minimum Password Strength: strong
Terms Required: yes
Terms Page: Not Set
Checkout Names: required
Checkout Address: required
Checkout Phone: optional
Checkout Email Confirmation: yes
Open Registration: yes
Registration Names: required
Registration Address: hidden
Registration Phone: hidden
Registration Voucher: hidden
Registration Email Confirmation: yes
Account Names: required
Account Address: required
Account Phone: optional
Account Email Confirmation: yes
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: PH
Currency: PHP
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: yes
Email From Address: [removed]
Email From Name: [removed]
Email Footer Text: 
Email Header Image: 
Cert Bg Width: 800
Cert Bg Height: 616
Cert Legacy Compat: no


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Stripe: Enabled
Stripe Test Mode: Enabled
Stripe Logging: yes
Stripe Order: 0
PayPal: Enabled
PayPal Test Mode: Enabled
PayPal Logging: yes
PayPal Order: 1
Manual: Enabled
Manual Logging: no
Manual Order: 2


Server
-------------------------------------------

Mysql Version: 5.7.37
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 5000
Php Max Upload Size: 512 MB
Php Memory Limit: 256M
Php Post Max Size: 1024M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 30
Php Version: 7.3.33
Software: Apache/2.4.52 (Unix) OpenSSL/1.1.1
Wp Memory Limit: 256M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.82 Safari/537.36


Theme
-------------------------------------------

Name: LifterLMS LaunchPad
Version: 2.4.9
Themeuri: https://lifterlms.com/launchpad/
Authoruri: https://gocodebox.com
Template: 
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

AutomatorWP: 2.0.1
AutomatorWP - LifterLMS: 1.0.2
Code Snippets: 2.14.3
Conditional Menus: 1.1.9
Elementor: 3.5.5
Elementor Pro: 3.6.0
Font Awesome: 4.1.1
LifterLMS: 5.9.0
LifterLMS Advanced Quizzes: 2.0.0
LifterLMS Advanced Videos: 1.0.0-beta.17
LifterLMS Custom Fields: 2.0.2
LifterLMS Groups: 1.0.0-beta.18
LifterLMS Labs: 1.6.0
LifterLMS MailChimp Add-on: 3.2.0
LifterLMS PayPal Gateway: 1.3.0
LifterLMS Private Areas: 1.1.5
LifterLMS Social Learning: 1.4.0
LifterLMS Stripe Payment Gateway: 5.4.0
LifterLMS Twilio: 1.1.2
LifterLMS WooCommerce: 2.2.1
Loco Translate: 2.5.8
User Switching: 1.5.8
WooCommerce: 6.2.0
WP Fusion: 3.38.42
WP Rollback: 1.7.1
Yoast SEO: 18.1


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No
LifterLMS Groups: Yes
Twilio: Yes
WooCommerce: No
LifterLMS MailChimp: Yes
LifterLMS Private Areas: Yes
LifterLMS Social Learning: Yes
Videos: Vimeo: Yes
Videos: Wistia: No
Videos: YouTube: Yes


Template Overrides
-------------------------------------------





```

</details>

This issue has be recreated:
+ [ ] Locally
+ [ ] On a staging site
+ [ ] On a production website
+ [ ] With only LifterLMS and a default theme

HS-186951

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description
Fixes #2005 
Also fixes the empty courses/membership condition, the checked input vals are empty arrays.

## How has this been tested?
manually

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by thomasplevy** [comment]
@eri-trabiccolo thanks for this quick turnaround... after reviewing it makes me wonder why this JS is even available on the post table page at all... I don't believe it should be loaded there, have you noticed or thought about this?

**Comment by eri-trabiccolo** [comment]
I did think about that and my theory was that:
a) you maybe wanted something like that, like a place where to put all the "admin js" related to the voucher post types. 
Or
b) unwanted, and there might be other cases like this to "fix".
So I said to myself either I ask why or I fix this and then will see.

**Comment by thomasplevy** [comment]
@eri-trabiccolo gotcha

The *reason* is oversight. The whole list of checks in this area should likely be checking the screen in addition to the post type instead of relying on `global $post_type` as that'll load them all on the post tables and, since they're all metabox JS, my guess is none of them have any post table functionality in them.

We can address this separately as now that i"m looking at it I'd like to take care of all of those files and only load them when they're actually needed. But this fix is good for now.

I'm going to open a new issue about the assest loading there.

File: .changelogs/vouchers-metabox.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+significance: patch
+type: fixed
+links: 
+- "#2005"
+entry: Fixed impossibility to search for vouchers in the voucher list table.
```
Comment (thomasplevy) [inline .changelogs/vouchers-metabox.yml]:
```suggestion
entry: Fixed an issue encountered when using the search box on the voucher admin posts list screen.
```

**Pull request merged**